### PR TITLE
feat: hide empty groups when searching on explore

### PR DESCRIPTION
**Pull request by rephus**
Closes: #3057

### Description:

Screenshot from 2022-08-25 11-11-46
Screenshot from 2022-08-25 11-11-52
Screenshot from 2022-08-25 11-12-00
Screenshot from 2022-08-25 12-14-15
Screenshot from 2022-08-25 12-14-24

### CODE REVIEW

**Comment by IrakliJani** [comment]
@rephus UX-wise hiding empty tables will make a big difference. That also gives us an opportunity to show an empty state when there are no matches

**Comment by TuringLovesDeathMetal** [comment]
+1 to what Irakli said. 

Maybe something like: `🔍 no results found`

For the empty state

**Comment by rephus** [comment]
> Maybe something like: 🔍 no results found


image

I did it already :) 

Do you want me to change the text / add the icon ?

**Comment by TuringLovesDeathMetal** [comment]
nope, this is perfect! 😍

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/TableTreeSections.tsx [hunk 1]
```diff
@@ -32,41 +32,110 @@ const TableTreeSections: FC<Props> = ({
     const sectionDepth = depth;
     const treeRootDepth = depth + 1;
     const hasNoMetrics = Object.keys(table.metrics).length <= 0;
+
+    const isSearching = !!searchQuery && searchQuery !== '';
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/TableTreeSections.tsx:36]: is this available from the underlying react query again?

Comment (owlas) [reply packages/frontend/src/components/Explorer/ExploreTree/TableTree/TableTreeSections.tsx:36]: Oh wait just read further down and this doesn't mean "is loading search results" it means "I have filtered to only see the results from a search"

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx [hunk 2]
```diff
@@ -101,6 +111,36 @@ const ExploreTree: FC<ExploreTreeProps> = ({
     const [search, setSearch] = useState<string>('');
     const [source, setSource] = useState<Source>();
 
+    const isSearching = !!search && search !== '';
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreTree/index.tsx:114]: this business logic is duplicated

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx [hunk 3]
```diff
@@ -101,6 +111,36 @@ const ExploreTree: FC<ExploreTreeProps> = ({
...
+            ...table.metrics,
+        });
+        const allFields = [...allValues, ...additionalMetrics].reduce<
+            Record<string, AdditionalMetric | Dimension | Metric>
+        >((acc, item) => ({ ...acc, [getItemId(item)]: item }), {});
+
+        return getSearchResults(allFields, search).size > 0;
+    };
+
+    const tableTrees = Object.values(explore.tables)
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreTree/index.tsx:127]: nice!

**Review by owlas — APPROVED** [review]
This is really smooth great job. 

Some inline comments for some nits I spotted. Not blocking merge

**Pull request merged**